src/Uno.UI/Helpers/WinUI/MathHelpers.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// inlined.cpp, math.cpp
+
+#nullable enable
+
+using Windows.Foundation;
+
+namespace Uno.UI.Helpers.WinUI
+{
+	internal static class MathHelpers
+	{
+		/// <summary>
+		/// Determines if two rectangles intersect.
+		/// </summary>
+		/// <param name="a">First rectangle.</param>
+		/// <param name="b">Second rectangle.</param>
+		/// <returns>True if rectangles intersect.</returns>
+		internal static bool DoRectsIntersect(Rect a, Rect b) =>
```

Comment:
I don't think this method at all does what its name and description say it does? It seems like it only returns true if Rect `a` is completely contained by and smaller than Rect `b`.

Reply:
This is according the WinUI implementation, but I think it works - it compares the left to right, top to bottom, etc. which seems to check that the "opposite" side is further, which seems to make sense when I sketched some examples

Reply:
you're right, I totally misread it.